presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ImplementIntersectAsUnion.java

```diff
@@ -0,0 +1,230 @@
+            FilterNode filterNode = createFilterNodeForIntersect(aggregationNode, aggregationsBuilder.build());
+
+            return new ProjectNode(idAllocator.getNextId(), filterNode, outputProjections.build());
+        }
+
+        private List<Symbol> generateOutputSybmbolsFrom(List<Symbol> sourceSymbols)
+        {
+            ImmutableList.Builder<Symbol> outputSymbolBuilder = ImmutableList.builder();
+            for (Symbol symbol : sourceSymbols) {
+                outputSymbolBuilder.add(allocateNewSymbolFrom(symbol));
+            }
+            return outputSymbolBuilder.build();
+        }
+
+        private Symbol allocateNewSymbolFrom(Symbol symbol)
+        {
+            return allocateNewSymbolFrom(symbol.getName(), symbolAllocator.getTypes().get(symbol));
+        }
+
+        private AggregationNode createAggregationNodeForIntersect(IntersectNode node, List<Symbol> sourceOutputSymbols, UnionNode sourceNode, ImmutableMap.Builder<Symbol, FunctionCall> aggregationsBuilder, ImmutableMap.Builder<Symbol, Expression> outputProjections)
+        {
+            ImmutableMap.Builder<Symbol, Signature> functionsBuilder = ImmutableMap.builder();
+            ImmutableList.Builder<Symbol> groupByKeys = ImmutableList.builder();
+
+            List<Symbol> intersectOutputSymbols = node.getOutputSymbols();
+            Signature countSignature = new Signature("count", AGGREGATE, "bigint", "boolean");
+
+            for (int i = intersectOutputSymbols.size(); i < sourceOutputSymbols.size(); i++) {
+                Symbol countSymbol = allocateNewSymbolFrom("count", BIGINT);
+                aggregationsBuilder.put(countSymbol, new FunctionCall(QualifiedName.of("count"), ImmutableList.<Expression>of(new QualifiedNameReference(sourceOutputSymbols.get(i).toQualifiedName()))));
+                functionsBuilder.put(countSymbol, countSignature);
+            }
+
+            for (Symbol nodeSymbol : intersectOutputSymbols) {
+                int fieldIndex = intersectOutputSymbols.indexOf(nodeSymbol);
+                Symbol outputSymbol = sourceOutputSymbols.get(fieldIndex);
+                groupByKeys.add(outputSymbol);
+                outputProjections.put(nodeSymbol, new QualifiedNameReference(outputSymbol.toQualifiedName()));
+            }
+
+            return new AggregationNode(idAllocator.getNextId(),
+                    sourceNode,
+                    groupByKeys.build(),
+                    aggregationsBuilder.build(),
+                    functionsBuilder.build(),
+                    ImmutableMap.<Symbol, Symbol>of(),
+                    ImmutableList.of(groupByKeys.build()),
+                    Step.SINGLE,
+                    Optional.empty(),
+                    1.0,
+                    Optional.empty());
+        }
+
+        private Symbol allocateNewSymbolFrom(String name, Type type)
```

Comment:
This method is unnecessary (it doesn't add much value). Just inline the call wherever it's used.